rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -2142,9 +2143,31 @@ rocalAudioFileSourceSingleShard(
                                                                                         context->master_graph->meta_data_reader()
                                                                                         );
         context->master_graph->set_loop(loop);
-        if(is_output) {
-            auto actual_output = context->master_graph->create_tensor(info, is_output);
-            context->master_graph->add_node<CopyNode>({output}, {actual_output});
+        if(downmix) {
+            TensorInfo output_info = info;
+            std::vector<size_t> output_dims = {context->user_batch_size(), info.dims()[1], 1};
+            // std::vector<size_t> output_dims;
+            // output_dims.resize(3);
+            // output_dims.at(0) = context->user_batch_size();
+            // output_dims.at(1) = info.dims()[1];
+            // output_dims.at(2) = 1;
+            output_info.set_dims(output_dims);
+            output_info.set_tensor_layout(RocalTensorlayout::NONE);
+            output_info.set_max_shape();
+
+            auto downmixed_output = context->master_graph->create_tensor(output_info, false);
```

Comment:
I have a doubt, If is_output is true can we not create_tensor with the second argument as true
Instead of adding another copy node?
@shobana-mcw Will that work?

Reply:
Update the output variable itself